src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultCSharpPublisher.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Text;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class DefaultCSharpPublisher : CSharpPublisher
+    {
+        private static readonly SourceText EmptySourceText = SourceText.From(string.Empty);
+        private readonly Dictionary<string, SourceText> _publishedSourceText;
+        private readonly Lazy<ILanguageServer> _server;
+        private readonly ForegroundDispatcher _foregroundDispatcher;
+        private ProjectSnapshotManagerBase _projectSnapshotManager;
+
+        public DefaultCSharpPublisher(
+            ForegroundDispatcher foregroundDispatcher,
+            Lazy<ILanguageServer> server)
+        {
+            if (foregroundDispatcher == null)
```

Comment:
`==` and `is` in the same file. This makes me 😢

Reply:
![image](https://media1.tenor.com/images/49d13a6c87bff90e4e46140a315b5c7a/tenor.gif)